File: Design and Programming Environments/Atletism GUI/src/main/java/service/UserService.java
package service;

import domain.User;
import domain.exceptions.ValidationException;
import domain.validator.UserValidator;
import repository.UserDbRepository;

public class UserService implements IService<User> {
    UserValidator validator;
    UserDbRepository reposiotry;

    public UserService(UserValidator val, UserDbRepository repo){
        this.validator = val;
        this.reposiotry = repo;
    }

    @Override
    public User findOne(int id) {
        return null;
    }

    @Override
    public User save(User entity) throws ValidationException {
        validator.validate(entity);
        return reposiotry.save(entity);
    }

    public boolean getLogin(User entity) throws ValidationException {
        return reposiotry.getLogin(entity);
    }
}

File: Design and Programming Environments/Atletism GUI .NET/Atletism GUI .NET/service/UserService.cs
using Atletism_GUI.NET.model;
using Atletism_GUI.NET.repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Atletism_GUI.NET.service
{
    public class UserService : IService<User>
    {
        private UserDbRepo repo;
        public UserService(UserDbRepo rep)
        {
            this.repo = rep;
        }
        public User findOne(int id)
        {
            throw new NotImplementedException();
        }

        public User save(User entity)
        {
            throw new NotImplementedException();
        }

        public bool getLogin(string username, string password)
        {
            if (repo.findBy(username, password) != null)
                return true;
            return false;
        }
    }
}
